myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+        """
+        Convert a current to A/F if its present unit isn't recommended.
+        """
+        recommended_units = [
+            myokit.parse_unit('pA/pF'),
+            myokit.parse_unit('uA/cm^2'),
+        ]
+
+        if var.unit() not in recommended_units:
+            self._convert_unit(var, 'A/F', helpers=helpers)
+
+    def _convert_potential_unit(self, var, helpers=None):
+        """
+        Convert a potential to mV if its present unit isn't recommended.
+        """
+        recommended_units = [myokit.units.mV]
+
+        if var.unit() not in recommended_units:
+            self._convert_unit(var, 'mV', helpers=helpers)
+
+    def _convert_unit(self, var, unit, helpers=None):
+        """
+        Convert a variable to the given unit if possible. Throws a warning if
+        the conversion is not possible.
+        """
+        if var.unit() is None:
+            return
+
+        try:
+            var.convert_unit(unit, helpers=helpers)
+        except myokit.IncompatibleUnitError:
+            warnings.warn(
+                'Unable to convert ' + var.qname() + ' to recommended'
+                ' units of ' + str(unit) + '.'
+            )
+
+    def _create_diffsl_variable_names(self, model):
+        """
+        Create diffsl-compatible names for all variables in the model.
+
+        The following strategy is followed:
+         - Fully qualified names are used for all variables.
+         - Variables are checked for special names, and changed if necessary.
+         - Unsupported characters like '.' and '_' are replaced.
+         - Any conflicts are resolved in a final step by appending a number.
+        """
+
+        # Detect names with special meanings
+        def special_start(name):
+            prefixes = ['dudt', 'F', 'G', 'in', 'M', 'out', 'u']
+            for prefix in prefixes:
+                if name.startswith(prefix + '_'):
+                    return True
+            return False
+
+        # Replacement characters for underscores and periods
+        uscore = 'Z'
+        period = 'Z'
+
+        # Create initial variable names
+        var_to_name = collections.OrderedDict()
+        for var in model.variables(deep=True, sort=True):
+            # Start from fully qualified name
+            name = var.qname()
+
+            # Avoid names with special meaning
+            if special_start(name):
+                name = 'var' + uscore + name
+
+            # Replace unsupported characters
+            name = name.replace('.', period).replace('_', uscore)
+
+            # Store (initial) name for var
+            var_to_name[var] = name
+
+        # Check for conflicts with known keywords
+        from . import keywords
+
+        needs_renaming = collections.OrderedDict()
+        for keyword in keywords:
+            needs_renaming[keyword] = []
+
+        pace = model.binding('pace')  # Reserve 'Vc' for pace variable
+        if pace is not None:
+            needs_renaming['Vc'] = []
```

Comment:
Is this because `Vc` is an input? Can we rewrite this to something more general please? E.g. letting the user specify a list of inputs, or maybe a single input for a pacing variable?